cmd/agent/agent.go

```diff
@@ -37,9 +44,36 @@ func NewAgentCmd() *cobra.Command {
 
 	agentCmd.AddCommand(startCmd)
 
+	agentCmd.AddCommand(runOnceCmd)
+
 	return agentCmd
 }
 
+func runOnce(cmd *cobra.Command, args []string) {
+	var err error
+
+	cfg, err := agent.DefaultConfig()
+	if err != nil {
+		log.Fatal("Failed to create the agent configuration: ", err)
+	}
+
+	cfg.DefinitionsPath = args[0]
+	cfg.ServiceName = serviceName
+	cfg.WebPort = port
+	cfg.TTL = TTL
+
+	a, err := agent.NewWithConfig(cfg)
+	if err != nil {
+		log.Fatal("Failed to create the agent: ", err)
+	}
```

Comment:
this appears to be a 1:1 copy of the same section in func start. not trying to be a pain in the ass, but is there a reason why you didn't split it in a small private function that does that and returns the cfg to call in both places?

Reply:
good point! Also, we don't need any of
```golang
cfg.ServiceName = serviceName
cfg.WebPort = port
cfg.TTL = TTL
```

In the `run-once` command.

Reply:
Now that I think about it, we don't actually need an `Agent` instance at all for this command, you just need to instantiate the `Checker`, and run it by invoking it (the checker itself is just a function).

So the command handler could be something as simple as:
```golang
checker, err := NewChecker(args[0])
// err handling
res, err := checker()
// err handling
res.PrettyPrint(os.Stdout) // or something like that
```

With this approach, my comment above about `agent.DoChecks` would not apply anymore.

Reply:
well, that opens the larger question on why we tag this onto the "agent" subcommand, it really is just a separate "let me debug this quickly" command. Eventually however we need to figure out how to do priv separation between checker and agent.

Reply:
Good point. For the time being, I would just put all the commands related to local execution on a target node to the `agent` group. As more sub-commands will spawn, we'll get a better idea on how to adjust that.

Reply:
I have removed the 3 commands 

```
cfg.ServiceName = serviceName
cfg.WebPort = port
cfg.TTL = TTL
```
And now it's only 3 command + error checks. I don't thinks it's worthy now to put them in a separate function, it would only unnecessary encomplicate it now.

Reply:
> Now that I think about it, we don't actually need an `Agent` instance at all for this command, we just need to instantiate the `Checker`, and run it by invoking it (the checker itself is just a function).
> 
> So the command handler could be something as simple as:
> 
> ```go
> checker, err := NewChecker(args[0])
> // err handling
> res, err := checker()
> // err handling
> res.PrettyPrint(os.Stdout) // or something like that
> ```
> 
> With this approach, my comment above about `agent.DoChecks` would not apply anymore.

And I have moved the DoCheck call inside the agent/agent.go. But there a problem arose, I don't no how to cope with it in go. It's an unused parameter. I did it in the python style

`_, err := agent.NewWithConfig(cfg)`

with the underdash, but it seems to fail in golang.